File: 프로그래머스/unrated/120890. 가까운 수/가까운 수.py
def solution(array, n):
    des = max(array[0] - n, n - array[0])
    answer = array[0]
    for arr in array:
        tmp = max(arr-n,n-arr)
        if des > tmp:
            des = tmp
            answer = arr
        elif des == tmp:
            answer = min(answer,arr)
    
    return answer

File: 프로그래머스/Lv.0/120890. 가까운 수/가까운 수.java
import java.lang.Math;

class Solution {
    public int solution(int[] array, int n) {
        int d = Integer.MAX_VALUE, answer = 0;
        
        for(int i=0;i<array.length;i++){
            int dev = Math.max(n-array[i], array[i]-n);
            if(d > dev){
                d = dev;
                answer = array[i];
            }
            else if(d == dev){
                answer = Math.min(answer, array[i]);
            }
        }
        
        return answer;
    }
}